custom_components/tuya_local/devices/madimack_elite_v3_heatpump_updated.yaml

```diff
@@ -0,0 +1,200 @@
+name: Madimack Elite V3 pool heatpump
+primary_entity:
+  entity: climate
+  dps:
+    - id: 101
+      name: hvac_mode
+      type: boolean
+      mapping:
+        - dps_val: false
+          value: "off"
+          icon: "mdi:hvac-off"
+        - dps_val: true
+          constraint: mode
+          conditions:
+            - dps_val: auto
+              icon: "mdi:refresh-auto"
+              value: auto
+            - dps_val: cold
+              icon: "mdi:snowflake"
+              value: cool
+            - dps_val: heating
+              icon: "mdi:hot-tub"
+              value: heat
+    - id: 102
+      name: mode
+      type: string
+      hidden: true
+    - id: 104
+      name: temperature
+      type: integer
+      mapping:
+        - constraint: temperature_unit
+          conditions:
+            - dps_val: f
+              range:
+                min: 60
+                max: 104
+      range:
+        min: 18
+        max: 40
+    # ^ in reality, the range is different for cool and auto
+    - id: 105
+      name: preset_mode
+      type: string
+      mapping:
+        - dps_val: silence
+          icon: "mdi:weather-night"

```

Comment:
You shouldn't try to set icons from both hvac_mode and preset_mode.  Just from one place is less complex.

Reply:
Understood - I'd like to try figure out how translation_key works to specify icons properly, but I'm having trouble finding documentation (I'm a HA newbie)

Reply:
There isn't much in the way of documentation, just the announcement here https://developers.home-assistant.io/blog/2024/01/19/icon-translations/ and official documentation here https://developers.home-assistant.io/docs/core/entity#icons

They document that state_attributes can be used to define icons for attributes like preset_mode, but don't go into detail about priorities when both state_attributes and state define icons.  But the default climate component defines icons for preset modes, so it must be handled.

I haven't looked at defining icons for climate entities yet, as the defaults are fairly comprehensive, but it is probably worth it for pool heatpumps.